src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/CollectionSortClauseBuilder.java

```diff
@@ -12,6 +21,107 @@ public CollectionSortClauseBuilder(CollectionSchemaObject collection) {
 
   @Override
   public SortClause buildAndValidate(ObjectNode sortNode) {
+    // $lexical is only special for Collections: handle first
+    JsonNode lexicalValue = sortNode.get(DocumentConstants.Fields.LEXICAL_CONTENT_FIELD);
+    if (lexicalValue != null) {
+      // We can also check if lexical sort supported by the collection:
+      if (!schema.lexicalConfig().enabled()) {
+        throw ErrorCodeV1.LEXICAL_NOT_ENABLED_FOR_COLLECTION.toApiException(
+            "Lexical search is not enabled for collection '%s'", schema.name());
+      }
+      if (sortNode.size() > 1) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE.toApiException(
+            "if sorting by '%s' no other sort expressions allowed",
+            DocumentConstants.Fields.LEXICAL_CONTENT_FIELD);
+      }
+      if (!lexicalValue.isTextual()) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE.toApiException(
+            "if sorting by '%s' value must be String, not %s",
+            DocumentConstants.Fields.LEXICAL_CONTENT_FIELD,
+            JsonUtil.nodeTypeAsString(lexicalValue));
+      }
+
+      return new SortClause(
+          Collections.singletonList(
+              SortExpression.collectionLexicalSort(lexicalValue.textValue())));
+    }
+
+    // Otherwise, use shared default processing
     return defaultBuildAndValidate(sortNode);
   }
+
+  @Override
+  protected String validateSortClausePath(String path) {
+    if (!NamingRules.FIELD.apply(path)) {
+      if (path.isEmpty()) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE_PATH.toApiException(
+            "path must be represented as a non-empty string");
+      }
+      // But allow "well-known" fields
+      switch (path) {
```

Comment:
nit: switch and if can be a single switch statement. use a guard on the case 

`case String s && s.isBlank():`